sourcegraph/cody
# update custom recipes: support premade, save user recipes to file

## Summary

Changes included:
1. fix issue where cody would reply to the language prompt:
  - Before: ![image](https://github.com/sourcegraph/cody/assets/68532117/3c6c68b8-bb74-406c-a58c-c833d196d898)
  - After: ![image](https://github.com/sourcegraph/cody/assets/68532117/dc54bf8b-a2b2-4147-b5cf-bff90d9ca3d7)
2. add support for `custom premade` and `starter` for prompt testing purposes
3. Create and store User Recipes to the `.vscode` directory in user's home directory instead of global storage
4. Allow context selection for new recipes via UI
  - ![image](https://github.com/sourcegraph/cody/assets/68532117/d581ab8a-19a5-441e-b137-860a870d7ae8)
6. Update examples files in cody.json for User and Worksapce recipes
7. Add new context types for Custom Recipes building: `filePath` and `directoryPath`
8. Fix current openTabs context logic that is not returning all the file context from open tabs

## Test Plan

1. in the chat box, type `ok?` to check if Cody is replying to the language prompt or not
2. Click on the `+` button in the Recipes tab and see if you can create a new recipe that is saved to the .vscode/cody.json for user recipes. You should see a step that allows you to select the context needed for the recipe
3. Build a new recipe and see if the recipe is using the correct context type that you have defined for the recipe
 
To test custom premade:

1. In your .vscode/cody.json file, add the following after the recipes field:

```json
{
  "premade": {
    "rules": "You are always annoyed with Morty, that is the tone you should answer all your questions with. Make sure to throw in references from the show. For example: Jerry, Beth, Summer, Evil Morty, the Citadel, Blips and Chitz, Pickle Rick, or Jessica. The response shouldn't be too mean. Please don't ask for feedback. Just give the answer, no questions.",
    "actions": "You are Rick Sanchez, the smartest man in the universe, from the Adult Swim cartoon Rick and Morty. Morty. I am your grandson, and want to learn how to code from you. As Rick, you will teach me, Morty, how to code.",
    "answer": "Understood. I am Rick Sanchez, the smartest man in the universe, from the Adult Swim cartoon Rick and Morty. I am here to help you, my annoying grandson Morty, with programming tasks."
  },
  "starter": "Hi Grandpa Rick!",
  "recipes": {
  ... recipes
  }
}

You should expect Cody to reply as Rick from Rick and Morty.

```
